### PR TITLE
Make PackageSpecRegistry its own packageRegistryOwner

### PR DESCRIPTION
**Pull request by elliottt**
Ensure that `<PackageSpecRegistry>` has its `packageRegistryOwner` field set to itself.

Ensuring that `packageRegistryOwner` gets propagated properly in the `<PackageSpecRegistry>` hierarchy.

### Test plan

More testing present in #9743

### CODE REVIEW

**Review by jez — APPROVED** [review]
This looks good but I think we can go one step further in this PR and change `ClassOrModuleRef::isPackageSpecSymbol` to take advantage of this new invariant, as both a performance improvement and an implicit test that this change working as intended.

**Comment by elliottt** [comment]
> This looks good but I think we can go one step further in this PR and change `ClassOrModuleRef::isPackageSpecSymbol` to take advantage of this new invariant, as both a performance improvement and an implicit test that this change working as intended.

Great idea!

File: core/Symbols.cc [hunk 1]
```diff
@@ -623,15 +623,7 @@ bool ClassOrModuleRef::isOnlyDefinedInFile(const GlobalState &gs, core::FileRef
...
-    while (sym.exists() && sym != core::Symbols::root()) {
-        if (sym == core::Symbols::PackageSpecRegistry()) {
-            return true;
-        }
-
-        sym = sym.data(gs)->owner;
-    }
-
-    return false;
+    return sym.exists() && sym.data(gs)->packageRegistryOwner == sym;
```
Comment (elliottt) [inline core/Symbols.cc:626]: Is this what you were thinking, @jez?

Comment (jez) [reply core/Symbols.cc:626]: beautiful

Comment (jez) [reply core/Symbols.cc:626]: Some edge cases to check: neither `::<root>` nor `::Test` should be `isPackageSpecSymbol`. You could write a test in `core/test/core_test.cc` to assert this if you wanted.
